## Week 7

In [1]:
import zipfile
import os 
from zipfile import ZipFile
import io

### reading zip files

In [7]:
path = os.getcwd()
path

'/Users/liamhettinger/07_Data_Wrangling'

In [3]:
# check what is in out folder
os.listdir(path + '/transaction_zips/')

['.DS_Store', 'file4.zip', 'file1.zip', 'file3.zip', 'file2.zip']

In [4]:
# create a clean list of zip  files
zip_files = []

for file_name in os.listdir(path + '/transaction_zips/'):
    if 'zip' in file_name:
        zip_files.append(file_name)

In [5]:
zip_files

['file4.zip', 'file1.zip', 'file3.zip', 'file2.zip']

In [6]:
with ZipFile('transaction_zips/' + 'file1.zip', 'r') as zf:
        # create a list of the zip contents
        zipped_files = zf.namelist()
        for file_name in zipped_files:
            print(file_name)
            input_file = zf.open(file_name, 'r') 
            # fix the encoding - we might need to test this out for windows users
            input_file = io.TextIOWrapper(input_file,encoding="utf-8")
            for line in input_file:
                print(line)

file1.csv
CustomerId,SalesOrderId,TransactionDate,GrandTotal,Status

1,36777536,6/1/21,180,fulfilled

2,36914274,6/1/21,119.94,fulfilled

3,37054577,6/1/21,119.95,returned

4,37137392,6/1/21,132.44,fulfilled

5,37075736,6/1/21,158.9,fulfilled


In [ ]:
zipped_files

In [ ]:
# let's just check one out
with ZipFile('transaction_zips/' + 'file1.zip', 'r') as zf:
        # create a list of the zip contents
        zipped_files = zf.namelist()
        # read in files
        for file_name in zipped_files:
            print(file_name)
            # open the compressed files
            input_file = zf.open(file_name, 'r') 
            # fix the encoding - we might need to test this out for windows users
            input_file = io.TextIOWrapper(input_file,encoding="utf-8")
            for line in input_file:
                print(line)

In [ ]:
# break out rooms - get the above code chunk going - loop through all zips


for zipped_file in zip_files:
    with ZipFile('transaction_zips/' + 'file1.zip', 'r') as zf:
        # create a list of the zip contents
        zipped_files = zf.namelist()
        # read in files
        for file_name in zipped_files:
            print(file_name)
            # open the compressed files
            input_file = zf.open(file_name, 'r') 
            # fix the encoding - we might need to test this out for windows users
            input_file = io.TextIOWrapper(input_file,encoding="utf-8")
            for line in input_file:
                print(line)
                    

### Regex

In [22]:
import re

In [ ]:
# match only checks the first part of the string
# search will look through the the entire string or just through different parts of it
string = 'something to check'
#print(re.match('check',string))
print(re.match('some',string))

In [ ]:
quote = 'it was the best of times, it was the worst of times'

In [ ]:
# establish the pattern and save it - this is a bit more programmatic 
best_pattern = re.compile(r'best')

In [ ]:
# call search() with the quote above

best_pattern.search(quote)

In [ ]:
# replace the pattern with a new phrase

best_pattern.sub(r'most okay',quote)

In [ ]:
it_pattern = re.compile(r'it')

## search only returns the first instance

#it_pattern.search(quote)

## sub will replace all instances 

it_pattern.sub('',quote)

In [ ]:
# breakout rooms


# ^ - matches if a string starts with a character
# $ - matches if a string ends with a character
# . - matches any character
# [] - matches if any characters are present
# + - matches one or more occurrences to the left
# ? - matches zero or one occurrence to the left
# | - used for alteration

# \A - matches if characters are at the beginning or end of the string
# \b - matches if characters are at the beginning or end of a word
# \d - matches any number
# \D - matches any non-numeric characters
# \s - matches any string that contains whitespace
# \w - matches any alphanumeric character

# \ - escapes any symbol you want to match
# [^ ] - the caret in brackets will reverse whatever pattern follows it 

In [ ]:

#pattern = re.compile(r'\be\w+')
pattern = re.compile(r'\be\w+')
string = 'this is an example  '
pattern.sub('answer', string)

In [ ]:
# brackets with a range of numbers will replace any numeric value
pattern = re.compile(r'[1-9]')
string = 'This is 1 example'
pattern.sub('X',string)

In [ ]:
# same with letters
pattern = re.compile(r'[a-jn-z]')
string = '123y5'
pattern.sub('4',string)

In [ ]:
# use \ to escape special characters
pattern = r'\$'
test_string = '$1'
result = re.match(pattern, test_string)
# check if result has match
if result:
    print(True)

In [ ]:
# matching the first and last letter of my name along with the right number or periods
pattern = r'^t....s$'
test_string = 'thomas'
result = re.match(pattern, test_string)
# check if result has match
if result:
    print(True)

In [ ]:
pattern = r'\d' # same as using [1-9]
test_string = '123'
result = re.match(pattern, test_string)
# check if result has match
if result:
    print(True)

In [ ]:
pattern = r'\w\s'
test_string = 'abc 123'
#test_string = ' a '
result = re.match(pattern, test_string)
# check if result has match
if result:
    print(True)

In [ ]:
# example of regex to remove punctuation
# \w matches any alphanumeric characte
# \s matches any whitespace
# inside of brackets, ^ means not 
test_string = 'as-df9.'
#punctuation_pattern = re.compile(r'[^\w\s]')
punctuation_pattern = re.compile(r'[^1-9]')
punctuation_pattern.sub('X',test_string)

In [ ]:
# breakout rooms


# ^ - matches if a string starts with a character
# $ - matches if a string ends with a character
# . - matches any character
# [] - matches if any characters are present
# + - matches one or more occurrences to the left
# ? - matches zero or one occurrence to the left
# | - used for alteration

# \A - matches if characters are at the beginning or end of the string
# \b - matches if characters are at the beginning or end of a word
# \d - matches any number
# \D - matches any non-numeric characters
# \s - matches any string that contains whitespace
# \w - matches any alphanumeric character

# \ - escapes any symbol you want to match
# [^ ] - the caret in brackets will reverse whatever pattern follows it  

import re

# Enter a pattern in re.compile() to change the below date to 06-01-2021
test_string = '06/01/2021'
pattern = re.compile(r'/')
pattern.sub('-',test_string)

# Enter a pattern in re.compile() to change bears and beets to be singular
test_string = "bears, beets, battlestar galactica"
punctuation_pattern = re.compile()
punctuation_pattern.sub('',test_string)

# Enter a pattern in re.compile() to strip everything but 'battlestar galactica' (don't just use a match on it)
test_string = "2 bears, 2 beets, battlestar galactica"
punctuation_pattern = re.compile()
punctuation_pattern.sub('',test_string)

In [ ]:
test_string = "2 bears, 2 beets, battlestar galactica"
punctuation_pattern = re.compile(r'\d\s\bb.....\s+')
punctuation_pattern.sub('',test_string)

### Brewery API

In [9]:
import requests
#https://shkspr.mobi/blog/2016/05/easy-apis-without-authentication/
#https://mixedanalytics.com/blog/list-actually-free-open-no-auth-needed-apis/

In [10]:
# brewery api 
# https://www.openbrewerydb.org/documentation/01-listbreweries
url = 'https://api.openbrewerydb.org/breweries?by_state=montana' # open this up in the browser

In [11]:
# save the request as a variable
r = requests.get(url)

In [12]:
r.status_code

200

In [13]:
# open that request and save it as a variable to work with later
brew_json = r.json()

In [14]:
from pprint import pprint

In [ ]:
# what data type is this
pprint(brew_json)

In [16]:
# print each item or print it out values accorring to key
for item in brew_json:
   # print(item)
    print(item['name'])     
    print(item['city'])

2 Basset Brewery
White Sulphur Springs
380 Brewing Company
Clancy
406 Brewing Company
Bozeman
Angry Hank's Microbrewery
Billings
Backslope Brewing
Columbia Falls
Bandit Brewing Co
Hamilton
Bar 3 BBQ and Brewing
Belgrade
Bayern Brewing Inc.
Missoula
Beaver Creek Brewery
Wibaux
Beaverhead Brewing Co
Dillon
Beehive Basin Brewery
Big Sky
Bias Brewing
Kalispell
Big Sky Brewing Co
Missoula
Bitter Root Brewing Co
Hamilton
Black Eagle Brewery
Black Eagle
Blackfoot River Brewing Co
Helena
Blacksmith Brewing Co
Stevensville
BLUE RIDGE BREWING
Malta
Bonsai Brewing Project
Whitefish
Bozeman Brewing Co
Bozeman


In [17]:
# pare down to just city and name
name_city_list = []
for item in brew_json:
    name_city_list.append({'name':item['name'],'city':item['city']})
    #for key in ('name','city'):
    #    name_city_list.append({key:item[key]})
        

In [18]:
# notice how long this is
name_city_list

[{'name': '2 Basset Brewery', 'city': 'White Sulphur Springs'},
 {'name': '380 Brewing Company', 'city': 'Clancy'},
 {'name': '406 Brewing Company', 'city': 'Bozeman'},
 {'name': "Angry Hank's Microbrewery", 'city': 'Billings'},
 {'name': 'Backslope Brewing', 'city': 'Columbia Falls'},
 {'name': 'Bandit Brewing Co', 'city': 'Hamilton'},
 {'name': 'Bar 3 BBQ and Brewing', 'city': 'Belgrade'},
 {'name': 'Bayern Brewing Inc.', 'city': 'Missoula'},
 {'name': 'Beaver Creek Brewery', 'city': 'Wibaux'},
 {'name': 'Beaverhead Brewing Co', 'city': 'Dillon'},
 {'name': 'Beehive Basin Brewery', 'city': 'Big Sky'},
 {'name': 'Bias Brewing', 'city': 'Kalispell'},
 {'name': 'Big Sky Brewing Co', 'city': 'Missoula'},
 {'name': 'Bitter Root Brewing Co', 'city': 'Hamilton'},
 {'name': 'Black Eagle Brewery', 'city': 'Black Eagle'},
 {'name': 'Blackfoot River Brewing Co', 'city': 'Helena'},
 {'name': 'Blacksmith Brewing Co', 'city': 'Stevensville'},
 {'name': 'BLUE RIDGE BREWING', 'city': 'Malta'},
 {'na

In [ ]:
import requests
# pagination
# let's loop through this - 
# the goal is to expand on this code chunk to append all pages of 
# montana breweries in {name,city} format to name_city_list. 
url = 'https://api.openbrewerydb.org/breweries?by_state=montana&page='
page_num = 1

name_city_list = []

r = requests.get(url + str(page_num))

brew_json = r.json()

for item in brew_json:
    name_city_list.append({'name':item['name'],'city':item['city']})
print(name_city_list)

In [ ]:
len(brew_json)

In [ ]:
len(name_city_list)

In [ ]:
name_city_list

In [ ]:
# breakout answer to pull all data from brewery api

name_city_list = []

url = 'https://api.openbrewerydb.org/breweries?by_state=montana&page='

for i in range(1,100):
    full_url = url + str(i) # create new url with page number
    print(full_url)
    
    r = requests.get(full_url)  # send request
    print(len(r.json()))
    if len(r.json()) == 0:
        break #break if the json is empty
    
    # here we will loop through each     
    for item in r.json():
        name_city_list.append({'name':item['name'],'city':item['city']})


In [ ]:
url = 'https://api.openbrewerydb.org/breweries?by_state=montana&page='
page_num = 1
name_city_list = []
'''
r = requests.get(url + str(page_num))

brew_json = r.json()
for item in brew_json:
    name_city_list.append({'name':item['name'],'city':item['city']})
    '''
while len(brew_json) > 0:
   
    r = requests.get(url + str(page_num))
    brew_json = r.json()
    for item in brew_json:
        name_city_list.append({'name':item['name'],'city':item['city']})
    page_num += 1
print(name_city_list)

In [ ]:
name_city_list

### Cleaning

Let's count how often certain words occur in brewery names across the state

What cleaning should we do?

* use regular expressions to remove brewery, brewing, co, company, microbrewery, brew
* use regular expressions to remove any punctuation
* strip any whitespace
* remove any duplicates
* change all names to lower case

If we go through these steps and create a list of clean brewery names, output should look like this:

['2 basset','380','406','angry hanks','backslope'...]

In [19]:
brewery = 'Conflux Brewing Co'

In [20]:
brewery.replace('Co','')

'nflux Brewing '

In [23]:
re.sub(r'Co\b','',brewery)

'Conflux Brewing '

In [24]:
name_list = []
for item in name_city_list:
    #print(item)
    brewery_name = item['name'].lower()
    brewery_name = re.sub(r'co\b','',brewery_name)
    brewery_name = re.sub(r'[^\w\s]','',brewery_name)
    name_list.append(brewery_name)

In [25]:
name_list

['2 basset brewery',
 '380 brewing company',
 '406 brewing company',
 'angry hanks microbrewery',
 'backslope brewing',
 'bandit brewing ',
 'bar 3 bbq and brewing',
 'bayern brewing inc',
 'beaver creek brewery',
 'beaverhead brewing ',
 'beehive basin brewery',
 'bias brewing',
 'big sky brewing ',
 'bitter root brewing ',
 'black eagle brewery',
 'blackfoot river brewing ',
 'blacksmith brewing ',
 'blue ridge brewing',
 'bonsai brewing project',
 'bozeman brewing ']

In [ ]:
clean_name_list = []
for name in name_list:
   # name = name.replace('brewery','').replace('company','').replace('brewing','').replace('brew','')
    name = re.sub(r'(brewery|company|brewing|brew|microbrewery)','',name)
    name = name.strip()
    name = name.replace('    ',' ')
    clean_name_list.append(name)

In [ ]:
len(clean_name_list)

In [ ]:
clean_name_list

In [ ]:
# remove duplicates
len(set(clean_name_list))

In [ ]:
clean_name_list = list(set(clean_name_list))

In [ ]:
# pull out words 
all_words = []
for brewery_name in clean_name_list:
    for word in brewery_name.split(" "):
        print(word)
        all_words.append(word)

In [ ]:
all_words

In [ ]:
import collections

In [ ]:
counter = collections.Counter(all_words)

In [ ]:
print(counter)

### Text Parsing & Sentiment Analysis

In [ ]:
nltk.download()

In [45]:
import nltk
from nltk.tokenize import sent_tokenize
from nltk.tokenize import word_tokenize
from nltk.probability import FreqDist
from nltk.corpus import stopwords
from nltk.stem import PorterStemmer
from nltk.stem.wordnet import WordNetLemmatizer
import string

In [46]:
with open('lotr','r') as file:
    text = file.read().replace('\n', '')

In [47]:
text

'When Mr. Bilbo Baggins of Bag End announced that he would shortly becelebrating his eleventy-first birthday with a party of specialmagnificence, there was much talk and excitement in Hobbiton.Bilbo was very rich and very peculiar, and had been thewonder of the Shire for sixty years, ever since his remarkabledisappearance and unexpected return. The riches he had brought backfrom his travels had now become a local legend, and it was popularlybelieved, whatever the old folk might say, that the Hill at Bag Endwas full of tunnels stuffed with treasure. And if that was not enoughfor fame, there was also his prolonged vigour to marvel at. Time woreon, but it seemed to have little effect on Mr. Baggins. At ninety hewas much the same as at fifty. At ninety-nine they began to call himwell-preserved; but unchanged would have been nearer the mark. Therewere some that shook their heads and thought this was too much of agood thing; it seemed unfair that anyone should possess (apparently)perpetual y

In [48]:
# tokenize sentences - create a list of sentences

tokenized_text = sent_tokenize(text)

print(tokenized_text)

['When Mr. Bilbo Baggins of Bag End announced that he would shortly becelebrating his eleventy-first birthday with a party of specialmagnificence, there was much talk and excitement in Hobbiton.Bilbo was very rich and very peculiar, and had been thewonder of the Shire for sixty years, ever since his remarkabledisappearance and unexpected return.', 'The riches he had brought backfrom his travels had now become a local legend, and it was popularlybelieved, whatever the old folk might say, that the Hill at Bag Endwas full of tunnels stuffed with treasure.', 'And if that was not enoughfor fame, there was also his prolonged vigour to marvel at.', 'Time woreon, but it seemed to have little effect on Mr. Baggins.', 'At ninety hewas much the same as at fifty.', 'At ninety-nine they began to call himwell-preserved; but unchanged would have been nearer the mark.', 'Therewere some that shook their heads and thought this was too much of agood thing; it seemed unfair that anyone should possess (app

In [49]:
# tokenize words
tokenized_word = word_tokenize(text)
print(tokenized_word)

['When', 'Mr.', 'Bilbo', 'Baggins', 'of', 'Bag', 'End', 'announced', 'that', 'he', 'would', 'shortly', 'becelebrating', 'his', 'eleventy-first', 'birthday', 'with', 'a', 'party', 'of', 'specialmagnificence', ',', 'there', 'was', 'much', 'talk', 'and', 'excitement', 'in', 'Hobbiton.Bilbo', 'was', 'very', 'rich', 'and', 'very', 'peculiar', ',', 'and', 'had', 'been', 'thewonder', 'of', 'the', 'Shire', 'for', 'sixty', 'years', ',', 'ever', 'since', 'his', 'remarkabledisappearance', 'and', 'unexpected', 'return', '.', 'The', 'riches', 'he', 'had', 'brought', 'backfrom', 'his', 'travels', 'had', 'now', 'become', 'a', 'local', 'legend', ',', 'and', 'it', 'was', 'popularlybelieved', ',', 'whatever', 'the', 'old', 'folk', 'might', 'say', ',', 'that', 'the', 'Hill', 'at', 'Bag', 'Endwas', 'full', 'of', 'tunnels', 'stuffed', 'with', 'treasure', '.', 'And', 'if', 'that', 'was', 'not', 'enoughfor', 'fame', ',', 'there', 'was', 'also', 'his', 'prolonged', 'vigour', 'to', 'marvel', 'at', '.', 'Time',

In [50]:
fdist = FreqDist(tokenized_word)
print(fdist)

<FreqDist with 646 samples and 1794 outcomes>


In [51]:
fdist.most_common(20)

[(',', 99),
 ('.', 72),
 ('and', 58),
 ('the', 55),
 ("''", 50),
 ('of', 41),
 ('``', 34),
 ('to', 28),
 ('a', 26),
 ('was', 26),
 ('he', 24),
 ('in', 21),
 (';', 20),
 ('his', 19),
 ('Mr.', 18),
 ('Bilbo', 18),
 ('I', 18),
 ('that', 17),
 ('as', 16),
 ('it', 15)]

In [52]:
stop_words = set(stopwords.words("english"))
print(stop_words)

{'am', 's', 'hers', 'was', 'ourselves', 'own', 'isn', 'o', 'our', "you'd", 'too', 'theirs', 'ain', 'should', 'needn', 'over', 'during', 'hadn', 'such', 'shouldn', "don't", 't', 'off', 'further', 'most', 'same', 'be', 'a', 'by', "you're", 'has', 'each', 'mustn', 'herself', 'me', 'weren', 'the', "haven't", 'up', 'again', 'then', 'my', "hadn't", 'this', 'an', 'about', 'through', 'd', "couldn't", 'other', 'she', "you've", 'ours', 'while', 'down', 'when', 'he', 'there', 'doing', 'didn', "needn't", 'or', "shouldn't", 'aren', "aren't", 'how', 'wasn', 'why', 'll', 'that', 'until', 'will', 'who', 'its', 'no', 'only', 'couldn', "shan't", 'it', 'do', "weren't", 'on', 'once', 'hasn', 'below', 'whom', 'been', 'between', 'to', 'above', 'wouldn', 'before', 'is', 'being', "won't", 'some', "doesn't", 'not', 're', 'after', 'yourselves', "she's", 'if', 'here', "you'll", 'because', "that'll", 'those', 'ma', 'him', 'were', 'itself', 'out', 'nor', 'does', 'very', "hasn't", 'm', 'you', 'but', 'are', 'against

In [55]:
more_punctuation = {"``","''"}

In [56]:
string.punctuation

'!"#$%&\'()*+,-./:;<=>?@[\\]^_`{|}~'

In [57]:
filtered_sent=[]
for w in tokenized_word:
    if w not in stop_words and w not in string.punctuation and w not in more_punctuation:
        w = w.lower()
        filtered_sent.append(w)
print(filtered_sent)

['when', 'mr.', 'bilbo', 'baggins', 'bag', 'end', 'announced', 'would', 'shortly', 'becelebrating', 'eleventy-first', 'birthday', 'party', 'specialmagnificence', 'much', 'talk', 'excitement', 'hobbiton.bilbo', 'rich', 'peculiar', 'thewonder', 'shire', 'sixty', 'years', 'ever', 'since', 'remarkabledisappearance', 'unexpected', 'return', 'the', 'riches', 'brought', 'backfrom', 'travels', 'become', 'local', 'legend', 'popularlybelieved', 'whatever', 'old', 'folk', 'might', 'say', 'hill', 'bag', 'endwas', 'full', 'tunnels', 'stuffed', 'treasure', 'and', 'enoughfor', 'fame', 'also', 'prolonged', 'vigour', 'marvel', 'time', 'woreon', 'seemed', 'little', 'effect', 'mr.', 'baggins', 'at', 'ninety', 'hewas', 'much', 'fifty', 'at', 'ninety-nine', 'began', 'call', 'himwell-preserved', 'unchanged', 'would', 'nearer', 'mark', 'therewere', 'shook', 'heads', 'thought', 'much', 'agood', 'thing', 'seemed', 'unfair', 'anyone', 'possess', 'apparently', 'perpetual', 'youth', 'well', 'reputedly', 'inexhaus

In [58]:
fdist = FreqDist(filtered_sent)
fdist.most_common(20)

[('mr.', 18),
 ('bilbo', 18),
 ('i', 18),
 ('old', 10),
 ('said', 10),
 ('bag', 9),
 ('end', 9),
 ('frodo', 9),
 ('baggins', 8),
 ('say', 7),
 ('and', 7),
 ('much', 6),
 ('come', 5),
 ('but', 5),
 ('live', 5),
 ('gaffer', 5),
 ('heard', 5),
 ('years', 4),
 ('the', 4),
 ('hill', 4)]

In [59]:
# stemming words
ps = PorterStemmer()

stemmed_words=[]
for w in filtered_sent:
    stemmed_words.append(ps.stem(w))

In [60]:
stemmed_words

['when',
 'mr.',
 'bilbo',
 'baggin',
 'bag',
 'end',
 'announc',
 'would',
 'shortli',
 'becelebr',
 'eleventy-first',
 'birthday',
 'parti',
 'specialmagnific',
 'much',
 'talk',
 'excit',
 'hobbiton.bilbo',
 'rich',
 'peculiar',
 'thewond',
 'shire',
 'sixti',
 'year',
 'ever',
 'sinc',
 'remarkabledisappear',
 'unexpect',
 'return',
 'the',
 'rich',
 'brought',
 'backfrom',
 'travel',
 'becom',
 'local',
 'legend',
 'popularlybeliev',
 'whatev',
 'old',
 'folk',
 'might',
 'say',
 'hill',
 'bag',
 'endwa',
 'full',
 'tunnel',
 'stuf',
 'treasur',
 'and',
 'enoughfor',
 'fame',
 'also',
 'prolong',
 'vigour',
 'marvel',
 'time',
 'woreon',
 'seem',
 'littl',
 'effect',
 'mr.',
 'baggin',
 'at',
 'nineti',
 'hewa',
 'much',
 'fifti',
 'at',
 'ninety-nin',
 'began',
 'call',
 'himwell-preserv',
 'unchang',
 'would',
 'nearer',
 'mark',
 'therewer',
 'shook',
 'head',
 'thought',
 'much',
 'agood',
 'thing',
 'seem',
 'unfair',
 'anyon',
 'possess',
 'appar',
 'perpetu',
 'youth',
 'we

In [ ]:
fdist = FreqDist(stemmed_words)
fdist.most_common(20)

In [ ]:
lem = WordNetLemmatizer()

lemmatized_words=[]
for w in filtered_sent:
    w = w.lower()
    lemmatized_words.append(lem.lemmatize(w,'v'))

In [ ]:
lemmatized_words

In [ ]:
fdist = FreqDist(lemmatized_words)
fdist.most_common(40)

In [ ]:
# read in lists of positive and negative words
positive_words = []
with open('positive-words.txt','r') as pos_file:
    for word in pos_file:
        word = word.strip()
        positive_words.append(word)

        
negative_words = []
with open('negative-words.txt','r') as neg_file:
    for word in neg_file:
        word = word.strip()
        negative_words.append(word)

In [ ]:
print(len(negative_words))
print(len(positive_words))

In [ ]:
sentiment_list = []

for word in lemmatized_words:
    if word in positive_words:
        sentiment_list.append({'word': word,'sentiment': 'positive'})
    elif word in negative_words:
        sentiment_list.append({'word': word,'sentiment': 'negative'})
    else:
        sentiment_list.append({'word': word,'sentiment': 'neutral'})


In [ ]:
sentiment_list

In [ ]:
for item in sentiment_list:
    if item['sentiment'] == 'positive':
        print(item['word'])

In [ ]:
for item in sentiment_list:
    if item['sentiment'] == 'negative':
        print(item['word'])